src/result_processor.c

```diff
@@ -245,61 +245,43 @@ ResultProcessor *RPScorer_New(const ExtScoringFunctionCtx *funcs,
 }
 
 /*******************************************************************************************************************
- *  Vector Similarity Result Processor
+ *  Additional Values Loader Result Processor
  *
  * It takes results from upstream (should be Index iterator or close; before any RP that need these field),
- * and add thier vecsim score to the right score field before sending them downstream.
+ * and add their additional value to the right score field before sending them downstream.
  *******************************************************************************************************************/
 
 typedef struct {
   ResultProcessor base;
-  const RLookupKey **keys;
-  size_t nkeys;
-} RPVecSim;
+} RPMetrics;
 
-static int rpvecsimNext(ResultProcessor *base, SearchResult *res) {
+static int rpMetricsNext(ResultProcessor *base, SearchResult *res) {
   int rc;
-  RPVecSim *self = (RPVecSim *)base;
 
   rc = base->upstream->Next(base->upstream, res);
   if (rc != RS_RESULT_OK) {
     return rc;
   }
 
-  // Add result to every score field.
-  // TODO: when we'll have a way to hold many results, add each result to its corresponding field
-  //  this will require scanning the entire IndexResult tree and looking for vector nodes whose score field name
-  //  stored in some entry of the self->keys array
-  RS_LOG_ASSERT(self->nkeys == 1, "Internal error, number of vector fields in a query is at most 1");
-  for (size_t i = 0; i < self->nkeys; i++) {
-    RSValue *val;
-    if (res->indexResult->type == RSResultType_HybridDistance) {
-      val = RS_NumVal(res->indexResult->agg.children[0]->dist.distance);
-    } else {
-      // The entire query is a TOP-K query, or this is hybrid query that doesn't use the doc score,
-      // so the distance is saved in the root of indexResult.
-      val = RS_NumVal(res->indexResult->dist.distance);
-    }
-    RLookup_WriteOwnKey(self->keys[i], &(res->rowdata), val);
```

Comment:
Why was this removed?

Reply:
This RP was replaced with a generic one - iterator that wants to export a metric write it directly to its `RSIndesResult`, and this RP is responsible for extracting the metric and insert it to the `RLookup` table (which is exactly what happened here before for the vecsim iterator)